tests/management/test_refresh_course_skills.py

```diff
@@ -194,12 +194,12 @@ def test_course_skill_not_saved_upon_exception(self,
                 [
                     '[TAXONOMY] Refresh Course Skills. Options: [%s]',
                     '[TAXONOMY] Refresh course skills process started.',
-                    f'[TAXONOMY] API Error for course_key: {self.course_1.key}',
-                    f'[TAXONOMY] API Error for course_key: {self.course_2.key}',
-                    '[TAXONOMY] Refresh course skills process completed. \n'
+                    '[TAXONOMY] API Error for key: {}'.format(self.course_1.key),
+                    f'[TAXONOMY] API Error for key: {self.course_2.key}',
+                    '[TAXONOMY] Refresh %s skills process completed. \n'
                     'Failures: %s \n'
-                    'Total Courses Updated Successfully: %s \n'
-                    'Total Courses Skipped: %s \n'
+                    'Total %s Updated Successfully: %s \n'
+                    'Total %s Skipped: %s \n'
```

Comment:
where is %s variable defined? Should this have a proper value here?

Reply:
its value is dependent on the `product_type` passed to `refresh_product_skill` method. The following line is refactored in this PR so that a genericness would be created.
https://github.com/openedx/taxonomy-connector/blob/master/taxonomy/utils.py#L198

Reply:
Yes, that is true. But in the tests, the messages being compared should have a valid value against %s, not hardcoded %s.

Reply:
i was of the same view while writing them but that's not how it works! I think it is resolved at the time of displaying them on screen otherwise they are treated as plain string 🤔 
e.g. https://github.com/openedx/taxonomy-connector/blob/master/tests/management/test_refresh_course_skills.py#L100